Content.Server/Tabletop/TabletopNanopoliaSetup.cs

```diff
@@ -0,0 +1,95 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Tabletop
+{
+    [UsedImplicitly]
+    public sealed partial class TabletopNanopoliaSetup : TabletopSetup
+    {
+
+        [DataField("FigurineCatPrototype", customTypeSerializer:typeof(PrototypeIdSerializer<EntityPrototype>))]
+        public string FigurineCatPrototype { get; private set; } = "FigurineCat";
+
+        [DataField("FigurineLaceupsPrototype", customTypeSerializer:typeof(PrototypeIdSerializer<EntityPrototype>))]
+        public string FigurineLaceupsPrototype { get; private set; } = "FigurineLaceups";
+
+        [DataField("FigurineMcgriffPrototype", customTypeSerializer:typeof(PrototypeIdSerializer<EntityPrototype>))]
+        public string FigurineMcgriffPrototype { get; private set; } = "FigurineMcgriff";
+
+        [DataField("FigurineTopHatPrototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+        public string FigurineTopHatPrototype { get; private set; } = "FigurineTopHat";
+
+        [DataField("FigurineGreenAutolatPrototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+        public string FigurineGreenAutolatPrototype { get; private set; } = "FigurineGreenAutolathe";
+
+        [DataField("FigurineRedProtolatdPrototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+        public string FigurineRedProtolatdPrototype { get; private set; } = "FigurineRedProtolathe";
+
+        public override void SetupTabletop(TabletopSession session, IEntityManager entityManager)
+        {
+            var board = entityManager.SpawnEntity(BoardPrototype, session.Position);
+
+            const float x1 = 7.25f;
+            const float x2 = 6.25f;
+
+            const float y1 = 7.25f;
+            const float y2 = 6.25f;
+
+            const float y3_1 = 5.25f;
+            const float y3_2 = 4.25f;
+
+            var center = session.Position;
+
+            // Cat pieces.
+            EntityUid tempQualifier = entityManager.SpawnEntity(FigurineCatPrototype, center.Offset(-x1, y1));
+            session.Entities.Add(tempQualifier);
+
+            // Laceup pieces.
+            EntityUid tempQualifier1 = entityManager.SpawnEntity(FigurineLaceupsPrototype, center.Offset(-x1, y2));
+            session.Entities.Add(tempQualifier1);
+
+            // Mcgriff pieces.
+            EntityUid tempQualifier2 = entityManager.SpawnEntity(FigurineMcgriffPrototype, center.Offset(-x2, y1));
+            session.Entities.Add(tempQualifier2);
+
+            // TopHat pieces.
+            EntityUid tempQualifier3 = entityManager.SpawnEntity(FigurineTopHatPrototype, center.Offset(-x2, y2));
+            session.Entities.Add(tempQualifier3);
+
+            // GreenAutolat pieces.
+            EntityUid tempQualifier4 = entityManager.SpawnEntity(FigurineGreenAutolatPrototype, center.Offset(-5.25f, y3_1));
```

Comment:
for all of these multiple-pieces use a for loop instead of copy paste N times bruh

Reply:
Sorry, but I don't know enough c# to write something myself

Reply:
for (float x = -5.25f; x <= end value; x += 1f)
{
    the copy pasted bit with x instead of the first number
}

Reply:
Like this?

EntityUid tempQualifier6 = entityManager.SpawnEntity(FigurineRedProtolatdPrototype, center.Offset(for (float x = -5.25f; x <= -3.25f; x += 1);, y3_1));
            session.Entities.Add(tempQualifier6);

Reply:
no the for loop goes outside, the code that is inside the brackets is executed multiple times
read something online if you dont know what it is :trollface:

Reply:
> no the for loop goes outside, the code that is inside the brackets is executed multiple times read something online if you dont know what it is :trollface:

I know a little about cycles. I just don't know English well and misunderstood the advice

Reply:
I figured out the loop (for) and did this